pkg/ruler/manager_test.go

```diff
@@ -98,6 +122,64 @@ func TestSyncRuleGroups(t *testing.T) {
 	})
 }
 
+func TestSyncRulesToManager(t *testing.T) {
```

Comment:
I'm not sure this test is much valuable as is, because doesn't give the confidence we want. This test looks intended to ensure `syncRulesToManager()` is concurrency safe (when you call it for different users), but we don't check that the actually configured rules for each tenant are correct: that's the missing bit to me. Also each tenant should have a different config (e.g. set a different `Name` and `Namespace` for each tenant's `RuleGroupDesc`).

Reply:
I just pushed another commit and TestSyncRulesToManager is removed. Instead I add a check on the synced rules created in file system, check the document is created in the dedicated folder and content seems ok for me. Could you give it another look? Thanks a lot for the details comments on my PR, it helps me a lot for understanding the code base.